packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js

```diff
@@ -6822,6 +6822,48 @@ QUnit.module('performance', () => {
         assert.ok($.isFunction(filter), 'filter is function');
     });
 
+    QUnit.test('tag labels should be correctly displayed with valueExpr as function and hideSelectedItems enabled (T1234032)', function(assert) {
+        const dataSource = [
+            { id: 1, scheme: 'schema 1', name: 'item1' },
+            { id: 2, scheme: 'schema 2', name: 'item2' },
+            { id: 3, scheme: 'schema 3', name: 'item3' },
+            { id: 4, scheme: 'schema 4', name: 'item4' },
+            { id: 5, scheme: 'schema 5', name: 'item5' },
+        ];
+
+        const $tagBox = $('#tagBox').dxTagBox({
+            dataSource,
+            valueExpr(x) {
+                return x && x.name + ' ' + x.scheme;
+            },
+            displayExpr: 'name',
+            hideSelectedItems: true,
+            opened: true,
+        });
+
+        const tagBox = $tagBox.dxTagBox('instance');
+        const list = tagBox._list;
+        const $list = list.$element();
+
+        $($list.find('.dx-list-item').eq(0)).trigger('dxclick');
+
+        tagBox.open();
+        $($list.find('.dx-list-item').eq(0)).trigger('dxclick');
+
+        tagBox.open();
+        $($list.find('.dx-list-item').eq(0)).trigger('dxclick');
+
+        tagBox.open();
+        $($list.find('.dx-list-item').eq(0)).trigger('dxclick');
+
+        tagBox.open();
+        $($list.find('.dx-list-item').eq(0)).trigger('dxclick');
+
+        const $tagContainer = $tagBox.find(`.${TAGBOX_TAG_CONTAINER_CLASS}`);
+
+        assert.strictEqual($.trim($tagContainer.text()), 'item1item2item3item4item5', 'label values are displayed correctly');
```

Comment:
just a question, why $.trim is needed here?

Reply:
I believe neighboring tests used this approach to check all tag labels in a single assert.

Example:
https://github.com/DevExpress/DevExtreme/blob/9227c7176ed9a46a00358285d3e197af042eaaf9/packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js#L1796

Reply:
ah, ok